src/formatters.jl

```diff
@@ -369,29 +369,32 @@ end
 
 function formatfigures(chunk, docformat::Pandoc)
     fignames = chunk.figures
+    length(fignames) > 0 || (return "")
+
     caption = chunk.options[:fig_cap]
+    label = get(chunk.options, :label, nothing)
     result = ""
     figstring = ""
     attribs = ""
     width = chunk.options[:out_width]
     height = chunk.options[:out_height]
 
     #Build figure attibutes
-    width == nothing || (attribs = "width=$width")
-    (attribs ≠ "" && height ≠ nothing ) && (attribs *= " ")
-    height == nothing   || (attribs *= "height=$height")
-    attribs == ""    || (attribs = "{$attribs}")
-    length(fignames) > 0 || (return "")
+    attribs = String[]
+    width == nothing  || push!(attribs, "width=$width")
+    height == nothing || push!(attribs, "height=$height")
+    label == nothing  || push!(attribs, "#fig:$label")
+    attribs = join(attribs, " ")
 
     if caption != nothing
-        result *= "![$caption]($(fignames[1]))$attribs\n"
+        result *= "![$caption]($(fignames[1])){$attribs}\n"
```

Comment:
What happens if attribs are empty? It would be better not to output empty `{}` although I'm not sure if that's a problem for pandoc.

Reply:
good question. I feel like I changed that locally but don't think I did. I'll update to form the attribs string with brackets (or empty, in the case of empty attribs)